corehq/apps/reports/filters/users.py

```diff
@@ -234,7 +241,8 @@ def selected_user_types(mobile_user_and_group_slugs):
         """
         usage: ``HQUserType.DEMO_USER in selected_user_types``
         """
-        return [int(t[3:]) for t in mobile_user_and_group_slugs
+
+        return [HQUserType.user_types_by_code[int(t[3:])].unfriendly_name for t in mobile_user_and_group_slugs
                 if t.startswith("t__") and t[3:].isdigit()]
```

Comment:
This changes the return value of `selected_user_types` from a list of codes to a list of `unfriendly_name`s.  I believe this will break existing references, since it's usually done as `HQUserType.DEMO_USER in selected_user_types`.

Luckily, `namedtuple`s behave like you'd expect in equality checks, so you could also just return a list of `HQUserType.SOMETHING` objects and the existing way those checks are written should keep working.

Reply:
oh yep. I had originally changed it from the index number to this, didn't realize that the named_tuple would break it where it's used: https://github.com/dimagi/commcare-hq/pull/22586/commits/ffbf2eff90f78065282bd3973529a89951b282fc